modules/tide_landing_page/src/Plugin/jsonapi/FieldEnhancer/BasicTextEnhancer.php

```diff
@@ -82,4 +83,73 @@ public function replaceUnicodeWhitespace(array $data) {
     return $data;
   }
 
+  public function addTableStylesToProcessed(&$data) {
+    // Check if 'value' and 'processed' keys exist in $data.
+    if (isset($data['value']) && isset($data['processed'])) {
+      // Load 'value' and 'processed' HTML as DOM objects for manipulation.
+      $valueDom = new \DOMDocument();
+      $processedDom = new \DOMDocument();
+
+      // Suppress warnings for malformed HTML in $value and $processed.
+      libxml_use_internal_errors(true);
+      $valueDom->loadHTML($data['value']);
+      $processedDom->loadHTML($data['processed']);
+      libxml_clear_errors();
+
+      // Get <table> elements from both value and processed DOMs.
+      $valueTables = $valueDom->getElementsByTagName('table');
+      $processedTables = $processedDom->getElementsByTagName('table');
+
+      // Apply data attributes and styles if there are tables in both DOMs.
+      if ($valueTables->length > 0 && $processedTables->length > 0) {
+        // Process the first <table> in both DOMs.
+        $valueTable = $valueTables->item(0);
+        $processedTable = $processedTables->item(0);
+        // Transfer the inline style from valueTable to processedTable.
+        if ($valueTable->hasAttribute('style')) {
+          $styleValue = $valueTable->getAttribute('style');
+          $processedTable->setAttribute('style', $styleValue);
+        }
+
+        // Process <col> elements in both DOMs.
+        $valueCols = $valueTable->getElementsByTagName('col');
+        $processedCols = $processedTable->getElementsByTagName('col');
+
+        for ($i = 0; $i < min($valueCols->length, $processedCols->length); $i++) {
+          $valueCol = $valueCols->item($i);
+          $processedCol = $processedCols->item($i);
+
+          if ($valueCol->hasAttribute('style')) {
+            // Parse the style attribute.
+            $styleValue = $valueCol->getAttribute('style');
+            $styles = explode(';', $styleValue);
+
+            foreach ($styles as $style) {
+              $style = trim($style);
+              if (!empty($style)) {
+                list($property, $value) = explode(':', $style, 2);
+                $property = trim($property);
+                $value = trim($value);
+
+                // Sanitize the property name for a valid data attribute.
+                $dataAttribute = 'data-' . str_replace([' ', '_'], '-', strtolower($property));
+                $processedCol->setAttribute($dataAttribute, $value);
+              }
+            }
+          }
+        }
+      }
+
+      // Extract only the inner HTML of the <body> tag, avoiding extra tags.
+      $processedBodyContent = '';
+      foreach ($processedDom->getElementsByTagName('body')->item(0)->childNodes as $node) {
```

Comment:
can this be simplified by using `DOMDocument::saveHTML` directly on the <body> element.
e.g.  
```php
$processedBodyContent = $processedDom->saveHTML($processedDom->getElementsByTagName('body')->item(0));
```